### PR TITLE
apps/scan: Enable PDI scanner eject pause feature

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

When a ballot is being accepted and a second ballot is inserted, we don't want the second ballot to be mistakenly sucked through the scanner. Though we have guards in place, if there's enough resistance against the first ballot, it's still possible that this can occur.

Luckily, the PDI scanner has an internal "eject pause" feature that is designed to detect a second ballot in the front and pause the ejection. It's more effective than our previous approach since it has lower level control over the motors and can act quicker.

This PR enables the eject pause feature (only for rear ejections - i.e. ballot
accepts) and adds the corresponding events for eject pause/eject resume.

## Demo Video or Screenshot

https://github.com/votingworks/vxsuite/assets/530106/b7a54dc6-d990-40ef-b42e-7b3b5b5ed8fa

Note that in the 4th scan in this video, there's a 2s delay before the "remove your ballot" message pops up. This is the case where the scanner status request times out (detailed in one of the commits). We could potentially change the status request timeout in this context to make this faster, but given that it's an extreme edge case requiring precise timing to reproduce, I opted to let it be slow and keep things simpler.

## Testing Plan
Manual testing, updated automated tests

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/pdi-scanner/src/rust/client.rs [hunk 1]
```diff
@@ -73,6 +74,28 @@ impl<T> Client<T> {
...
+    // delay sending a response to a command (e.g. when the "eject pause"
+    // feature pauses the scanner, it will queue commands received during that
+    // time).
+    fn clear_unhandled_packets_except_events(&mut self) {
+        let unhandled_non_event_packets = self
+            .unhandled_packets
+            .iter()
+            .filter(|packet| !packet.is_event())
+            .collect::<Vec<_>>();
+        if unhandled_non_event_packets.len() > 0 {
```
Comment (eventualbuddha) [inline libs/pdi-scanner/src/rust/client.rs:93]: Per clippy:

```suggestion
        if !unhandled_non_event_packets.is_empty() {
```

**Review by eventualbuddha — APPROVED** [review]

File: libs/pdi-scanner/src/rust/client.rs (hunk 1 above)
Comment (jonahkagan) [reply libs/pdi-scanner/src/rust/client.rs]: thanks, changed.

maybe we should run clippy automatically?
